extension/xhprof.c

```diff
@@ -1109,9 +1109,16 @@ void hp_sample_check(hp_entry_t **entries  TSRMLS_DC) {
 static inline uint64 cycle_timer() {
   uint32 __a,__d;
   uint64 val;
+#if defined(__x86_64__)
   asm volatile("rdtsc" : "=a" (__a), "=d" (__d));
   (val) = ((uint64)__a) | (((uint64)__d)<<32);
   return val;
+#elif defined(__ARM_ARCH)
+  struct timeval now;
+  if (gettimeofday(&now, NULL) == 0) {
```

Comment:
Hmm, why not use this on x64 as well? gettimeofday is a POSIX API IIRC.

Reply:
I'm assuming the assembly is faster, as it's operating on the CPU, not via an OS sys call?

Reply:
Right. It looks like rdtsc is getting the clock cycle count, so this is actually just an approximation of that to emulate it; it's then undone later in `get_tsc_from_us` based on `get_cpu_frequency`.

Given we're fixing the "frequency" here, we should also ifdef in get_cpu_frequency and return a static constant as we know what it is, which will remove a 5ms sleep delay on startup; alternatively, these CPU cycle functions could all be ifdef'd out, and we could just use gettimeofday directly if it's fast enough in the places where this is used.